tools/update_ruby_gem_packages.rb

```diff
@@ -5,55 +5,63 @@
 # tools/update_ruby_gem_packages.rb
 
 # Add >LOCAL< lib to LOAD_PATH
-$LOAD_PATH.unshift '../lib'
+$LOAD_PATH.unshift './lib'
+
+require 'json'
+require 'net/http'
 require_relative '../lib/color'
 require_relative '../lib/const'
-require_relative '../lib/gem_compact_index_client'
+require_relative '../lib/package'
+require_relative '../lib/package_utils'
 require_relative '../lib/require_gem'
-require_gem('activesupport', 'active_support/core_ext/object/blank')
 require_gem 'concurrent-ruby'
+require_gem 'ruby-libversion', 'ruby_libversion'
 
 def check_for_updated_ruby_packages
-  gems = BasicCompactIndexClient.new.gems
-
+  # Create a thread pool for parallelization.
   pool = Concurrent::ThreadPoolExecutor.new(
     min_threads: 1,
     max_threads: CREW_NPROC.to_i + 1,
     max_queue: 0, # unbounded work queue
     fallback_policy: :caller_runs
   )
+
+  # Currently, the only packages that use the ruby buildsystem are ruby_* packages, so this finds all the packages we need to check.
   relevant_gem_packages = Dir['packages/ruby_*.rb']
+
+  # Get the total number of files to check, and then the length of that number, so status updates can be formatted.
   total_files_to_check = relevant_gem_packages.length
   numlength = total_files_to_check.to_s.length
   updateable_packages = {}
+  packages_without_gem_versions = []
   relevant_gem_packages.each_with_index do |package, index|
     pool.post do
-      untested_package_name = package.gsub(%r{^packages/ruby_}, '').gsub(/.rb$/, '')
-      gem_test = gems.grep(/#{"^#{untested_package_name}\\s.*$"}/).last.blank? ? gems.grep(/#{"^\(#{passed_name.gsub(/^ruby_/, '').gsub('_', ')*.(')}\\s\).*$"}/).last : gems.grep(/#{"^#{untested_package_name}\\s.*$"}/).last
-      abort "Cannot find #{passed_name} gem to install.".lightred if gem_test.blank?
-      gem_test_name = gem_test.split.first
-      gem_test_versions = gem_test.split[1].split(',')
-      # Any version with a letter is considered a prerelease as per
-      # https://github.com/rubygems/rubygems/blob/b5798efd348935634d4e0e2b846d4f455582db48/lib/rubygems/version.rb#L305
-      gem_test_versions.delete_if { |i| i.match?(/[a-zA-Z]/) }
-      gem_test_version = gem_test_versions.max
-      gem_name = gem_test_name.blank? ? Gem::SpecFetcher.fetcher.suggest_gems_from_name(untested_package_name).first : gem_test_name
-      gem_version = gem_test_name.blank? ? Gem.latest_version_for(untested_package_name).to_s : gem_test_version
-      next package if gem_version.blank?
-
-      relevant_gem_packages.delete(package)
+      pkg = Package.load_package(package)
+      gem_name = pkg.name.sub('ruby_', '').gsub('_', '-')
```

Comment:
This doesn't replace the logic above, which is there for a reason: some gems have a mix of underscores and dashes in their names!

Reply:
Ah. Some packages do benefit from this logic, like rubocop-chromebrew, so I'll see about setting up some special cases.

Reply:
How about?
```
gem_name = pkg.name.sub('ruby_', '').gsub('_', '-') unless pkg.name.include?('-') && pkg.name.include?('_')
```

Reply:
oooh

Reply:
Fixed! I had to use manual special cases, because we don't actually know whether the gem originally used underscores or dashes. Still, we now check every package:
```
Done checking rubygems for updates to 133 ruby packages.
Updated versions were not listed in rubygems for: 
```